### PR TITLE
Mldbfb 634 join structured column

### PR DESCRIPTION
**Pull request by steadtler**
Added python converter for rows of rows

Normal executor:

Needed to check both the simplified prefix and the initial prefix because normal tables dont return the table name, but in joins we do return the name of the child table as part of the column name.

Pipeline executor:

Needed to pass the table name in the getallcolumns (wildcard) so that the join knows which of its child table is concerned

Needed to evaluate each atom by the wildcard filter

### CODE REVIEW

File: plugins/lang/python/mldb_python_converters.h [hunk 1]
```diff
@@ -122,6 +120,71 @@ struct CellValueConverter
...
+    static void construct(PyObject* obj_ptr, void* storage)
+    {
+        auto pyToPathElement = [] (PyObject* ptr)
+        {
+            if (PyInt_Check(ptr)) {
+                int val = boost::python::extract<int>(ptr);
+                return PathElement(std::to_string(val));
+            }
+            else if (PyFloat_Check(ptr)) {
+                double val = boost::python::extract<double>(ptr);
```
Comment (FinchPowers) [inline plugins/lang/python/mldb_python_converters.h]: Bien vu. ;)

Comment (mailletf) [inline plugins/lang/python/mldb_python_converters.h]: :P

File: sql/execution_pipeline_impl.cc [hunk 2]
```diff
@@ -105,47 +106,35 @@ doGetAllColumns(std::function<ColumnName (const ColumnName &)> keep,
...
 
-            const ExpressionValue & rowContents
-            = row.values.at(fieldOffset + ROW_CONTENTS);
+    auto exec = [=] (const SqlRowScope & rowScope, const VariableFilter & filter) -> ExpressionValue
+    {
+        auto & row = rowScope.as<PipelineResults>();
 
-            RowValue result;
+        const ExpressionValue & rowContents
+        = row.values.at(fieldOffset + ROW_CONTENTS);
```
Comment (FinchPowers) [inline sql/execution_pipeline_impl.cc]: indent

File: sql/execution_pipeline_impl.cc [hunk 3]
```diff
@@ -617,61 +607,82 @@ doGetColumn(const ColumnName & columnName, int fieldOffset)
...
         rightPrefix = right->as();
-    
-    auto leftOutput = left->doGetAllColumns(keep, leftFieldOffset(fieldOffset));
-    auto rightOutput = right->doGetAllColumns(keep, rightFieldOffset(fieldOffset));
+
+    bool useLeft = tableName.empty() || tableName == leftPrefix;
+    bool useRight = tableName.empty() || tableName == rightPrefix;
+
+    if (useLeft)
+        cerr << "use left" << endl;
```
Comment (FinchPowers) [inline sql/execution_pipeline_impl.cc]: use the logging interface

**Comment by FinchPowers** [comment]
cerr à enlever / remplacer et +1

**Pull request merged**